packages/@lwc/template-compiler/src/codegen/index.ts

```diff
@@ -723,12 +729,13 @@ function generateTemplateFunction(codeGen: CodeGen): t.FunctionDeclaration {
     );
 }
 
-export default function (root: Root, state: State): string {
+export default function (root: Root, state: State, apiVersion: APIVersion): string {
```

Comment:
Same here, `state` has the apiVersion. Not require to pass it as a separate value.